website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -1184,6 +1184,45 @@ func init() {
 
 </span>
 
+{{< paragraph class="language-python">}}
+Proper Use of return vs yield in Python Functions.
+In Python, functions can return results either all at once (return) or lazily one at a time (yield). The right choice depends on your use case, especially when dealing with large datasets or streaming scenarios.
```

Comment:
We don't need to include information on how Python generally handles return vs yield, we should specifically talk about how Beam handles them